tests/security/pam/pam_login.pm

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: PAM tests for login, user login should fail without authentication
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#70345, tc#1767577
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Define the user and password, which are already configured in previous milestone
+    my $user   = 'suse';
+    my $passwd = 'susetesting';
+
+    # Modify the login/sshd files to set the PAM authentication
+    my $deny_user_file = '/etc/deniedusers';
+    my $pam_sshd       = '/etc/pam.d/sshd';
+    my $pam_login      = '/etc/pam.d/login';
+    my $pam_sshd_bak   = '/tmp/sshd_bak';
+    my $pam_login_bak  = '/tmp/login_bak';
+    assert_script_run "echo $user > $deny_user_file";
+    assert_script_run "cp $pam_sshd $pam_sshd_bak";
+    assert_script_run "cp $pam_login $pam_login_bak";
+    assert_script_run "sed -i '\$a auth      required   pam_listfile.so   onerr=succeed  item=user  sense=deny  file=$deny_user_file' $pam_sshd";
+    assert_script_run "sed -i '\$a auth      required   pam_listfile.so   onerr=succeed  item=user  sense=deny  file=$deny_user_file' $pam_login";
+
+    # Try to login to the OS with user suse, access should fail
+    assert_script_run(
+        "expect -c 'spawn ssh $user\@localhost; \\
+expect \"Password: \"; send \"$passwd\\n\"; \\
+expect \"Password: \"; send \"$passwd\\n\"; \\
+expect \"Password: \"; send \"$passwd\\n\"; \\
+expect \"*password: \"; send \"$passwd\\n\"; \\
+expect \"*password: \"; send \"$passwd\\n\"; interact'"
+    );
+
```

Comment:
The expected messages are all `*assword:` it seems not explicitly show the access should be failed.
E.g., `Permission denied, please try again.`

Reply:
I will modify the code

Reply:
I am considering to catch the logs, thanks

Reply:
I used ssh command here, then if it passed, it will switch to a new bash interface, so adding the check code should not work here.

But later code will check the current user.

Reply:
I mean to expect "Permission denied, please try again." will be better since ssh will be failed.

Reply:
hmm, let me have a try